include/deal.II/base/job_identifier.h

```diff
@@ -66,21 +66,19 @@ class JobIdentifier
   const std::string
   operator()() const;
 
+  /**
+   * Function to identify the presently running program.
+   */
+  static JobIdentifier &
+  get_dealjobid();
+
 private:
   /**
    * String holding the identifier of the presently running program.
    */
   std::string id;
 };
 
-
-/*----------------------------- Inline functions ----------------------------*/
-
-
-/**
- * Global object to identify the presently running program.
- */
-extern JobIdentifier dealjobid;
```

Comment:
this is strictly speaking an incompatible change. I think it's so obscure that there isn't even a point in mentioning that :-)

Reply:
Yes, you are right, of course.